src/System/Net/HttpListener.cs

```diff
@@ -27,10 +28,11 @@ public sealed unsafe partial class HttpListener : IDisposable
         private AuthenticationSchemes _authenticationScheme = AuthenticationSchemes.Anonymous;
         private ExtendedProtectionSelector? _extendedProtectionSelectorDelegate;
         private string? _realm;
+        private X509Certificate2? _certificate;
 
         internal ICollection PrefixCollection => _uriPrefixes.Keys;
 
-        public HttpListener()
+        public HttpListener(X509Certificate2? certificate = null)
```

Comment:
Just don't take the value via constructor at all, users should just call `SetCertificate()` later.

Reply:
Ahhh I see now, yeah it will fine too that way. Will do it asap